docs/setup/config.md

```diff
@@ -272,7 +272,9 @@ For example when running Flink on YARN on an environment with a restrictive fire
 
 - `recovery.zookeeper.quorum`: Defines the ZooKeeper quorum URL which is used to connet to the ZooKeeper cluster when the 'zookeeper' recovery mode is selected
 
-- `recovery.zookeeper.path.root`: (Default '/flink') Defines the root dir under which the ZooKeeper recovery mode will create znodes.
+- `recovery.zookeeper.path.root`: (Default '/flink') Defines the root dir under which the ZooKeeper recovery mode will create namespace directories.
+
+- `recovery.zookeeper.path.namespace`: (Default '/default_ns' in standalone mode, or the <yarn-application-id> under Yarn) Defines the subdirectory under the root dir where the ZooKeeper recovery mode will create znodes. This allows to isolate multiple applications on the same ZooKeeper. 
```

Comment:
I think the main problem with default namespaces in standalone mode is that there is no authority generating reliably unique identifier that we can use to label clusters (in contrast to e.g. application ids in Yarn). Also the contents of the masters file could be the same for two different clusters running on the same machines and collisions could happen. In particular, such collisions could be very rare and hence even more surprising to the user. Furthermore, I think that having multiple clusters in this way is exactly when you want to start Yarn or Mesos.